wagtail/users/utils.py

```diff
@@ -25,24 +27,37 @@ def user_can_delete_user(current_user, user_to_delete):
     return True
 
 
-def get_gravatar_url(email, size=50):
-    default = "mm"
-    size = (
-        int(size) * 2
-    )  # requested at retina size by default and scaled down at point of use with css
+def get_gravatar_url(email, size=50, default_params={}):
+    DEFAULT_IMAGE = "mp"
+    DEFAULT_GRAVATAR_PROVIDER_URL = "www.gravatar.com/avatar"
+
+    params = {
+        "s": int(size)
+        * 2,  # requested at retina size by default and scaled down at point of use with css
+        "d": DEFAULT_IMAGE,
+    }
+
     gravatar_provider_url = getattr(
-        settings, "WAGTAIL_GRAVATAR_PROVIDER_URL", "//www.gravatar.com/avatar"
+        settings, "WAGTAIL_GRAVATAR_PROVIDER_URL", DEFAULT_GRAVATAR_PROVIDER_URL
     )
 
     if (not email) or (gravatar_provider_url is None):
         return None
 
-    email_bytes = email.lower().encode("utf-8")
-    hash = safe_md5(email_bytes, usedforsecurity=False).hexdigest()
-    gravatar_url = "{gravatar_provider_url}/{hash}?{params}".format(
-        gravatar_provider_url=gravatar_provider_url.rstrip("/"),
-        hash=hash,
-        params=urlencode({"s": size, "d": default}),
+    parsed_url = urlparse(gravatar_provider_url.rstrip("/"))
+
+    if parsed_url.query:
+        params.update(
+            **{key: value[-1] for key, value in parse_qs(parsed_url.query).items()},
+        )
+    params.update(**default_params)
```

Comment:
Should we override default_params by provider url query's parameters? 

i thought that provider url is defined in settings means its global so if someone who wants different default avatar then it can be achieved by passing defaul_params in the argument but overriding it with provider url's qp will never let do that if there is already d in provider url's qp

Reply:
The function won't normally get used directly, it's an escape hatch for more advanced customisations. 

I would expect the default would act as a default, as in when other values are provided in the URL configured in the settings, these would override the default.

Reply:
Just another thought, we should ensure that the size param (s) gets added last. We don't want this to be overridden by the provided default params or the URL. 

What do you think?

Reply:
Yes, it makes sense to override the default parameters with the size argument since it's passed separately. I realized this after you added that code snippet (thank you for that!), I have made the final changes, so please let me know if I have gone in the wrong direction.